File: VB/DiagramDataBindingBehavior/ViewModel/ViewModel.vb
Imports DiagramDataBindingBehavior.Data
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace DiagramDataBindingBehavior.ViewModel

	Public Class ViewModel
		Public Property ClassSource() As List(Of ClassData)
		Public Property ConnectionSource() As List(Of ConnectionData)

		Public Sub New()
			ClassSource = ClassStructureGenerator.ClassList()
			ConnectionSource = ClassStructureGenerator.ConnectionList()
		End Sub
	End Class
End Namespace

File: CS/DiagramDataBindingBehavior/ViewModel/ViewModel.cs
using DiagramDataBindingBehavior.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DiagramDataBindingBehavior.ViewModel
{
    
    public class ViewModel
    {
        public List<ClassData> ClassSource { get; set; }
        public List<ConnectionData> ConnectionSource { get; set; }

        public ViewModel()
        {
            ClassSource = ClassStructureGenerator.ClassList();
            ConnectionSource = ClassStructureGenerator.ConnectionList();
        }
    }
}
